hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -896,6 +907,40 @@ public void reclaimPreviouslyReservedThrottle(int n, HederaFunctionality functio
         synchronizedThrottleAccumulator.leakUnusedThrottlePreviouslyReserved(n, function);
     }
 
+    @Override
+    public boolean shouldThrottleNOfUnscaled(int n, HederaFunctionality function) {
+        return synchronizedThrottleAccumulator.shouldThrottleNOfUnscaled(n, function, userTransactionConsensusTime);
+    }
+
+    public boolean shouldThrottleTxn(TransactionInfo txInfo) {
+        return networkUtilizationManager.shouldThrottle(txInfo, current());
+    }
+
+    @Override
+    public boolean hasThrottleCapacityForChildTransactions() {
+        var isAllowed = true;
+        final var childRecords = recordListBuilder.childRecordBuilders();
+
+        for (int i = 0, n = childRecords.size(); i < n && isAllowed; i++) {
+            final var childRecord = childRecords.get(i);
+            if (Objects.equals(childRecord.status(), SUCCESS)) {
+                final var tx = childRecord.build().transaction();
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏Not sure if we don't need a new getter for the `transaction` instead of calling `build()` 🤔.